### ISSUE TITLE
CI/CD ?

### ISSUE THREAD

**Issue by volth** [nix-windows/nix#28]
Is there a free (or reasonably cheap) CI/CD with Windows and caching proxy (so we won't have to download gigabytes from `msys.org` and `microsoft.com` on each run) ?

Or setting that on prem is the only viable option ?

**Comment by Ericson2314** [comment]
I was thinking of borrowing some techniques from the CI for Meson itself, https://github.com/mesonbuild/meson/blob/master/azure-pipelines.yml. I might still download tons of stuff each time, but at least I think it's free.

**Comment by AtnNn** [comment]
I've been experimenting with a Hydra instance at https://thanos.atnnn.com/project/nix-windows

It's not a Windows builder, but I was able to test a few related scenarios:
- I failed to evaluare `nixpkgs-master` for a `cygwin` system (https://thanos.atnnn.com/jobset/nix-windows/nixpkgs-cygwin#tabs-errors)
- I failed to evaluate `nixpkgs-windows` for a `windows` system (https://thanos.atnnn.com/jobset/nix-windows/nixpkgs-windows-0.5#tabs-errors)
- Some `pkgsCross.mingW64` packages that used to work don't build or evaluate anymore (https://thanos.atnnn.com/jobset/nix-windows/nixpkgs-master-cross-mingwW64#tabs-jobs)
- `nix-windows` doesn't build on Linux (https://thanos.atnnn.com/build/15353)

Ideally I would like to add a Windows host as a remote builder, but as far as I can tell:
- `nix-windows` doesn't support that yet
- Having an incompatible nix store path (`/nix/store` vs `c:\nix\store`) makes it impossible

If I can get a connect a Cygwin or MSYS2 build of nix as a remote builder, I'll attempt to build MSVC2019 packages with it.

**Comment by Ericson2314** [comment]
Yeah I definitely want to dogfood too, but I think some non-Nix CI would be practical in the meantime.

**Comment by colemickens** [comment]
I was wondering if [`wfvm`](https://git.m-labs.hk/M-Labs/wfvm) could help assist with this goal.

**Comment by jared-w** [comment]
Github Actions is free for open source repositories and comes with quite a few things bundled into its windows images; so you might not need to cache as many things. It also has support for windows containers through docker (downloading a large container and extracting is much faster than downloading packages and installing). Lastly, I think Github Action's cache would be large enough that you could cache everything (either 5GB per repo or per org, iirc).

It's worth checking out, at least? It's also quite similar to azure devops pipelines so you can borrow inspiration between the two fairly easily.

**Comment by volth** [comment]
> It's worth checking out, at least?

GitHub is a social network first with all the problems of a social network (deplatforming accounts at random, etc)
